### PR TITLE
Pass Flang libraries to MSVC linker correctly

### PR DESCRIPTION
**Pull request by kaadam**
The orignal PR was created by Isuruf #65
This change are missing from the recent version of Classic-LLVM.

### CODE REVIEW

**Comment by kaadam** [comment]
Hi @bryanpkc could you look at this PR, please? May I ask you to add this PR to the list of the LLVM 16 changes? If it is too late, I can rebase it after the merge. :) Thanks.

File: clang/lib/Driver/ToolChains/CommonArgs.h
Comment (bryanpkc) [inline clang/lib/Driver/ToolChains/CommonArgs.h:27]: Move this up two lines so that it is guarded by `ENABLE_CLASSIC_FLANG`.

File: clang/lib/Driver/ToolChains/MSVC.cpp
Comment (bryanpkc) [inline clang/lib/Driver/ToolChains/MSVC.cpp:508]: Leave a blank line between this and the previous function.

**Comment by bryanpkc** [comment]
> May I ask you to add this PR to the list of the LLVM 16 changes?

Let's merge this into `release_15x` first, then you can create a new PR for `release_16x`. My changes for LLVM 16 are still blocked by a libomptarget test failure, but I hope to resolve that this weekend.

File: clang/lib/Driver/ToolChains/CommonArgs.h
Comment (kaadam) [reply clang/lib/Driver/ToolChains/CommonArgs.h:27]: Done.

File: clang/lib/Driver/ToolChains/MSVC.cpp
Comment (bryanpkc) [inline clang/lib/Driver/ToolChains/MSVC.cpp]: Do we need `libompstub` here? `ToolChain::AddFortranStdlibLibArgs` doesn't add it any more.

Comment (kaadam) [reply clang/lib/Driver/ToolChains/MSVC.cpp]: Yes, good point. It could be removed since we can use MS its own version of libomp.

**Review by bryanpkc — APPROVED** [review]
LGTM

**Comment by bryanpkc** [comment]
Please create a PR targeting `release_16x`.

**Comment by kaadam** [comment]
> Please create a PR targeting `release_16x`.

Thanks for the review Bryan. I've opened a PR for release_16x.

**Review by pawosm-arm — APPROVED** [review]

**Pull request merged**
